war/src/main/less/base/typography.less

```diff
@@ -51,45 +51,45 @@ h6,
   line-height: var(--line-height-heading);
   font-weight: bold;
   display: block;
-  margin-top: 0.5rem;
-  margin-bottom: 0.5rem;
+  margin-top: 0;
+  margin-bottom: var(--section-padding);
```

Comment:
@janfaracik This change seems intended to fix some particular problem, but it's unclear which problem it was intended to fix and in fact it causes a new problem as described in JENKINS-69549 and #7078. In order to ensure that the fix for that bug doesn't re-expose whatever problem this original change was intended to fix, it would be helpful to know what specifically this original change was intended to fix.